browser/ui/webui/brave_rewards_internals_ui.cc

```diff
@@ -401,6 +414,53 @@ void RewardsInternalsDOMHandler::OnGetEventLogs(ledger::type::EventLogs logs) {
       std::move(data));
 }
 
+void RewardsInternalsDOMHandler::GetAdDiagnostics(const base::ListValue* args) {
+  if (!ads_service_) {
+    NOTREACHED();
+    return;
+  }
+
+  ads_service_->GetAdDiagnostics(
+      base::BindOnce(&RewardsInternalsDOMHandler::OnGetAdDiagnostics,
+                     weak_ptr_factory_.GetWeakPtr()));
+}
+
+void RewardsInternalsDOMHandler::OnGetAdDiagnostics(const bool success,
+                                                    const std::string& json) {
+  constexpr char kName[] = "name";
+  constexpr char kValue[] = "value";
+
+  if (!web_ui()->CanCallJavascript()) {
+    return;
+  }
+
+  base::Value diagnostics(base::Value::Type::LIST);
+  base::Value status(base::Value::Type::DICTIONARY);
+  status.SetStringKey(kName, "Status");
+  status.SetStringKey(kValue, "Ads are not initialized yet. Press Refresh button please.");
```

Comment:
Wouldn't we want to localize these messages?

Reply:
Done